src/main/java/org/jenkinsci/plugins/saml/SamlSecurityRealm.java

```diff
@@ -331,7 +332,7 @@ public HttpResponse doFinishLogin(final StaplerRequest2 request, final StaplerRe
 
         SamlAuthenticationToken samlAuthToken = new SamlAuthenticationToken(userDetails);
 
-        ACL.as2(samlAuthToken);
+        SecurityContextHolder.setContext(new SecurityContextImpl(samlAuthToken));
```

Comment:
```suggestion
        SecurityContextHolder.getContext().setAuthentication(samlAuthToken);
```

as in for example https://github.com/jenkinsci/oic-auth-plugin/blob/6bf419104767e4b285f33834351e9c08fec0bf37/src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java#L1056 https://github.com/jenkinsci/github-oauth-plugin/blob/697aa152ab3641728b2c917534b42240b3bd638e/src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java#L413 https://github.com/jenkinsci/gitlab-oauth-plugin/blob/f5e0680072e5b35337d2070e46610d5f1be626ca/src/main/java/org/jenkinsci/plugins/GitLabSecurityRealm.java#L384 https://github.com/jenkinsci/azure-ad-plugin/blob/da1c9ebef38623746076a10270417e09c1114f79/src/main/java/com/microsoft/jenkins/azuread/AzureSecurityRealm.java#L476 https://github.com/jenkinsci/keycloak-plugin/blob/11575f0480436f4feafc409423cd6f6f1d0df7c6/src/main/java/org/jenkinsci/plugins/KeycloakSecurityRealm.java#L268

Reply:
done , I have re tested , everything is working fine